library/src/main/java/com/bekawestberg/loopinglayout/library/LoopingLayoutManager.kt

```diff
@@ -155,82 +149,54 @@ class LoopingLayoutManager : LayoutManager, RecyclerView.SmoothScroller.ScrollVe
                 ViewGroup.LayoutParams.WRAP_CONTENT)
     }
 
-    override fun onLayoutChildren(recycler: RecyclerView.Recycler, state: RecyclerView.State) {
-        if (mPendingScrollPosition != RecyclerView.NO_POSITION) {
-            layoutToPosition(recycler, state)
-            return
+    override fun onSaveInstanceState(): Parcelable? {
+        val direction = getMovementDirectionFromAdapterDirection(TOWARDS_LOWER_INDICES)
+        return LayoutRequest(
+                anchorIndex = getInitialIndex(direction),
+                scrollOffset = getInitialItem(direction).hiddenSize)
+    }
+
+    override fun onRestoreInstanceState(state: Parcelable?) {
+        if (state is LayoutRequest) {
+            layoutRequest = state
         }
-        layoutAnew(recycler, state)
     }
 
-    /**
-     * Lays out the views by scrapping the current ones (if any exist) and starting from scratch.
-     */
-    private fun layoutAnew(recycler: RecyclerView.Recycler, state: RecyclerView.State) {
+    override fun onLayoutChildren(recycler: RecyclerView.Recycler, state: RecyclerView.State) {
+        layoutRequest.initialize(this, state);
+
         detachAndScrapAttachedViews(recycler)
         var layoutRect = nonScrollingEdges
 
-        val movementDir = getMovementDirectionFromAdapterDirection(TOWARDS_HIGHER_INDICES)
+        // The initial index should be laid out at the edge associated with the adapter index, so
```

Comment:
```suggestion
        // The initial index should be laid out at the edge associated with the adapter direction, so
```

Or see if there is a more clear way to phrase this (ascii?)